cmd/corimDisplay.go

```diff
@@ -61,72 +61,119 @@ func checkCorimDisplayArgs() error {
+		// convert []corim.Tag to [][]byte
+		tags := make([][]byte, len(s.UnsignedCorim.Tags))
+		for i, tag := range s.UnsignedCorim.Tags {
+			tags[i] = tag
+		}
+		displayTags(tags)
 	}
 
-	fmt.Println("Meta:")
-	fmt.Println(string(metaJSON))
+	return nil
+}
 
-	if corimJSON, err = json.MarshalIndent(&s.UnsignedCorim, "", "  "); err != nil {
-		return fmt.Errorf("error decoding unsigned CoRIM from %s: %w", signedCorimFile, err)
+func displayUnsignedCorim(u corim.UnsignedCorim, corimFile string, showTags bool) error {
+	corimJSON, err := json.MarshalIndent(&u, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding unsigned CoRIM from %s: %w", corimFile, err)
 	}
 
 	fmt.Println("Corim:")
 	fmt.Println(string(corimJSON))
 
 	if showTags {
 		fmt.Println("Tags:")
-		for i, e := range s.UnsignedCorim.Tags {
-			// need at least 3 bytes for the tag and 1 for the smallest bstr
-			if len(e) < 3+1 {
-				fmt.Printf(">> skipping malformed tag at index %d\n", i)
-				continue
-			}
-
-			// split tag from data
-			cborTag, cborData := e[:3], e[3:]
-
-			hdr := fmt.Sprintf(">> [ %d ]", i)
-
-			if bytes.Equal(cborTag, corim.ComidTag) {
-				if err = printComid(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoMID tag at index %d: %v\n", i, err)
-				}
-			} else if bytes.Equal(cborTag, corim.CoswidTag) {
-				if err = printCoswid(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoSWID tag at index %d: %v\n", i, err)
-				}
-			} else if bytes.Equal(cborTag, cots.CotsTag) {
-				if err = printCots(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoTS tag at index %d: %v\n", i, err)
-				}
-			} else {
-				fmt.Printf(">> unmatched CBOR tag: %x\n", cborTag)
-			}
+		// convert []corim.Tag to [][]byte
+		tags := make([][]byte, len(u.Tags))
+		for i, tag := range u.Tags {
+			tags[i] = tag
 		}
+		displayTags(tags)
 	}
 
 	return nil
 }
 
+func display(corimFile string, showTags bool) error {
+	var (
+		corimCBOR []byte
+		err       error
+	)
+
+	// read the CoRIM file
+	if corimCBOR, err = afero.ReadFile(fs, corimFile); err != nil {
+		return fmt.Errorf("error loading CoRIM from %s: %w", corimFile, err)
+	}
+
+	// try to decode as a signed CoRIM
+	var s corim.SignedCorim
+	if err = s.FromCOSE(corimCBOR); err == nil {
+		// successfully decoded as signed CoRIM
+		return displaySignedCorim(s, corimFile, showTags)
+	}
+
+	// if decoding as signed CoRIM failed, attempt to decode as unsigned CoRIM
+	var u corim.UnsignedCorim
+	if err = u.FromCBOR(corimCBOR); err != nil {
+		return fmt.Errorf("error decoding CoRIM (signed or unsigned) from %s: %w", corimFile, err)
+	}
+
+	// successfully decoded as unsigned CoRIM
+	return displayUnsignedCorim(u, corimFile, showTags)
+}
+
+// displayTags processes and displays embedded tags within a CoRIM
+func displayTags(tags [][]byte) {
+	for i, e := range tags {
+		// ensure the tag has at least 4 bytes (3 for tag identifier and 1 for data)
```

Comment:
The description of the constant’s significance should be above its definition as a named constant. Then change the literal to the named constant.